src/frontend/src/binder/expr/function.rs

```diff
@@ -62,6 +62,10 @@ impl Binder {
                     inputs = Self::rewrite_nullif_to_case_when(inputs)?;
                     ExprType::Case
                 }
+                "concat_ws" => {
+                    inputs = Self::rewrite_concat_ws_args_when(inputs)?;
+                    ExprType::ConcatWs
```

Comment:
Just return `ExprType::ConcatWs` here and add casting in `FunctionCall::new`. cast-only changes do not use dedicated `rewrite_xxx` functions.